.github/actions/test-setup/action.yml

```diff
@@ -0,0 +1,38 @@
+name: "Setup Python and Virtual Environment"
+description: "Sets up Python, a virtual environment, and installs dependencies"
+
+inputs:
+  python-version:
+    description: "Python version to use"
+    required: true
+  venv-dir:
+    description: "Virtual environment directory"
+    required: true
+    default: "${{ github.workspace }}/.venv"
+
+runs:
+  using: "composite"
+  steps:
+    - name: Get Current Date
+      id: date
+      shell: bash
+      run: echo "date=$(date +'%Y-%m-%d')" >> $GITHUB_ENV
+
+    - name: Set up Python
+      uses: actions/setup-python@v5.4.0
+      with:
+        python-version: ${{ inputs.python-version }}
+
+    - name: Create Python Virtual Environment
+      shell: bash
+      run: python -m venv ${{ inputs.venv-dir }}
+
+    - name: Install Dependencies
```

Comment:
You might want to cache those.

Reply:
Could you please elaborate little more with examples?